test/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.FunctionalTests/ApplicationWithParseErrorsTest.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.ViewCompilation
+{
+    public class ApplicationWithParseErrorsTest : IClassFixture<ApplicationWithParseErrorsTest.ApplicationWithParseErrorsFixture>
+    {
+        public ApplicationWithParseErrorsTest(ApplicationWithParseErrorsFixture fixture)
+        {
+            Fixture = fixture;
+        }
+
+        public ApplicationWithParseErrorsFixture Fixture { get; }
+
+        public static TheoryData SupportedFlavorsTheoryData => RuntimeFlavors.SupportedFlavorsTheoryData;
+
+        [Theory]
+        [MemberData(nameof(SupportedFlavorsTheoryData))]
+        public void PublishingPrintsParseErrors(RuntimeFlavor flavor)
```

Comment:
Turns out this is hard to test - `Location.FilePath` is used specifically for imported files (i.e. _ViewImports). This works during regular compilation because we compile one view at a time. In the tool, we compile in parallel. The razor host caches parsed content, so it ends up being a race. I'm going to just use the FullPath for now.